x-pack/solutions/observability/plugins/apm/server/types.ts

```diff
@@ -64,6 +64,7 @@ import type { APMConfig } from '.';
 
 export interface APMPluginSetup {
   config$: Observable<APMConfig>;
+  config: APMConfig;
```

Comment:
Exposing the APM config in order to get access to the `managedServiceUrl` property in a more convenient way instead of getting it from the observable. Not sure if this is an an anti-pattern though, happy to change if needed.

Reply:
It looks strange at first, tbh. In your use case, it wouldn't be an anti-pattern per se because you're using the `currentConfig` in the `observability_onboarding` server `setup`. I've seen other plugins exposing the synchronous config object, but it should be used with caution.

Perhaps we could rename this variable to `configInitialValue`, to make it more explicit that this object won't reflect changes made to the config, or instead of exposing this, you could do:

```ts
const managedServiceUrl = await firstValueFrom(plugins.apm.setup.config$).then((config) => {
  return config.managedServiceUrl;
});
```

Reply:
Thanks @crespocarlos! Looking at it again, I think I'll follow your suggestion to just use `firstValueFrom` because the observable is already there, and while the code is a bit cumbersome, the managed service URL will be exposed through Cloud plugin soon anyway, so I'll just refactor it later.